### PR TITLE
Iroha Permissions

### PR DESCRIPTION
**Pull request by humb1t**

### CODE REVIEW

File: iroha/src/asset.rs [hunk 1]
```diff
@@ -45,6 +47,18 @@ impl Asset {
             quantity,
             big_quantity: 0,
```
Comment (kamilsa) [inline iroha/src/asset.rs:48]: why do we need both quantity and big_quantity?

Comment (humb1t) [reply iroha/src/asset.rs:48]: @takemiyamakoto asked about separation of this two values, but I don't think that we will save any memory by this - good question, made an issue for this: https://soramitsu.atlassian.net/browse/RH2-102

File: iroha/src/isi.rs [hunk 2]
```diff
@@ -21,9 +21,14 @@ pub enum Instruction {
...
     Account(crate::account::isi::AccountInstruction),
-    /// This variant of Iroha Special Instructions composes two other instructions into one, and
+    /// Variant of instructions related to `Permission`.
+    Permission(crate::permission::isi::PermissionInstruction),
+    /// This variant of Iroha Special Instruction composes two other instructions into one, and
     /// executes them both.
     Compose(Box<Instruction>, Box<Instruction>),
+    /// This variant of Iroha Special Instruction executes the first instruction and if it succeeded
+    /// executes the second one, if failed - the third one.
+    If(Box<Instruction>, Box<Instruction>, Box<Instruction>),
```
Comment (eadventurous) [inline iroha/src/isi.rs:31]: Maybe make second and third instructions optional? Sometimes it makes sense to execute instruction only on success and also sometimes only on failure.

We can also imitate this behavior with `DoNothing` instruction.

Comment (humb1t) [reply iroha/src/isi.rs:31]: Will align set of instruction to needs of users

**Pull request merged**